encfs/main.cpp

```diff
@@ -768,9 +772,18 @@ static bool unmountFS(EncFS_Context *ctx) {
     return false;
   } else {
     // Time to unmount!
-    RLOG(WARNING) << "Unmounting filesystem due to inactivity: "
-                  << arg->opts->mountPoint;
-    fuse_unmount(arg->opts->mountPoint.c_str());
-    return true;
+    int rc=1;
+    if (access("/bin/umount", F_OK) != -1)
+      rc = system(("/bin/umount "+std::string(arg->opts->mountPoint)).c_str());
+    else
+      rc = system(("/sbin/umount "+std::string(arg->opts->mountPoint)).c_str());
```

Comment:
> - we can't call `fuse_unmount` as we don't have a fuse handle (`fuse_main` does not return it) ;

Well, according to encfs4win call to `fuse_unmount` (second parameter to `NULL`), and according to [libfuse source code](https://github.com/libfuse/libfuse/blob/fuse_2_9_5/lib/helper.c#L250), we could clearly set this second parameter to `NULL`.

Let me do some tests, however I think I will still face the drawbacks listed in the first post.